ci/rtools/awssdk_ep.patch

```diff
@@ -0,0 +1,213 @@
+--- a/aws-cpp-sdk-core/source/http/windows/WinHttpSyncHttpClient.cpp
++++ b/aws-cpp-sdk-core/source/http/windows/WinHttpSyncHttpClient.cpp
+@@ -22,6 +22,16 @@
+ #include <sstream>
+ #include <iostream>
+
++#ifndef WINHTTP_OPTION_WEB_SOCKET_KEEPALIVE_INTERVAL
++#define WINHTTP_OPTION_WEB_SOCKET_KEEPALIVE_INTERVAL 116
++#endif
++#ifndef WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_1
++#define WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_1 0x00000200
++#endif
++#ifndef WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_2
++#define WINHTTP_FLAG_SECURE_PROTOCOL_TLS1_2 0x00000800
++#endif
++
+ using namespace Aws::Client;
+ using namespace Aws::Http;
+ using namespace Aws::Http::Standard;
+@@ -272,7 +282,7 @@ bool WinHttpSyncHttpClient::DoQueryHeaders(void* hHttpRequest, std::shared_ptr<H
+     wmemset(contentTypeStr, 0, static_cast<size_t>(dwSize / sizeof(wchar_t)));
+
+     WinHttpQueryHeaders(hHttpRequest, WINHTTP_QUERY_CONTENT_TYPE, nullptr, &contentTypeStr, &dwSize, 0);
+-    if (contentTypeStr[0] != NULL)
++    if (contentTypeStr[0])
+     {
+         Aws::String contentStr = StringUtils::FromWString(contentTypeStr);
+         response->SetContentType(contentStr);
+@@ -303,7 +313,7 @@ bool WinHttpSyncHttpClient::DoQueryHeaders(void* hHttpRequest, std::shared_ptr<H
+
+ bool WinHttpSyncHttpClient::DoSendRequest(void* hHttpRequest) const
+ {
+-    return (WinHttpSendRequest(hHttpRequest, NULL, NULL, 0, 0, 0, NULL) != 0);
++    return (WinHttpSendRequest(hHttpRequest, NULL, 0, 0, 0, 0, 0) != 0);
+ }
+
+ bool WinHttpSyncHttpClient::DoReadData(void* hHttpRequest, char* body, uint64_t size, uint64_t& read) const
+diff --git a/aws-cpp-sdk-core/source/http/windows/WinINetSyncHttpClient.cpp b/aws-cpp-sdk-core/source/http/windows/WinINetSyncHttpClient.cpp
+index 5854cc334..70184a358 100644
+--- a/aws-cpp-sdk-core/source/http/windows/WinINetSyncHttpClient.cpp
++++ b/aws-cpp-sdk-core/source/http/windows/WinINetSyncHttpClient.cpp
+@@ -225,7 +225,7 @@ bool WinINetSyncHttpClient::DoQueryHeaders(void* hHttpRequest, std::shared_ptr<H
+     char contentTypeStr[1024];
+     dwSize = sizeof(contentTypeStr);
+     HttpQueryInfoA(hHttpRequest, HTTP_QUERY_CONTENT_TYPE, &contentTypeStr, &dwSize, 0);
+-    if (contentTypeStr[0] != NULL)
++    if (contentTypeStr[0])
+     {
+         response->SetContentType(contentTypeStr);
+         AWS_LOGSTREAM_DEBUG(GetLogTag(), "Received content type " << contentTypeStr);
+diff --git a/aws-cpp-sdk-core/source/http/windows/WinSyncHttpClient.cpp b/aws-cpp-sdk-core/source/http/windows/WinSyncHttpClient.cpp
+index d7513cc3c..e390a8d4e 100644
+--- a/aws-cpp-sdk-core/source/http/windows/WinSyncHttpClient.cpp
++++ b/aws-cpp-sdk-core/source/http/windows/WinSyncHttpClient.cpp
+@@ -349,7 +349,7 @@ std::shared_ptr<HttpResponse> WinSyncHttpClient::MakeRequest(const std::shared_p
+         }
+     }
+
+-    if (!success && !IsRequestProcessingEnabled() || !ContinueRequest(*request))
++    if (!success && (!IsRequestProcessingEnabled() || !ContinueRequest(*request)))
```

Comment:
Are we sure this is the right parenthesis placement? :) Personally, I can't tell just from the diff...

Reply:
Great catch, I think you're right that I made the wrong change, see https://github.com/aws/aws-sdk-cpp/blob/main/src/aws-cpp-sdk-core/source/http/windows/WinSyncHttpClient.cpp#L406 where it's,

```
    if ((!success && !IsRequestProcessingEnabled()) || !ContinueRequest(*request))
```

Reply:
Fixed in 069413831e9adbb5a994336606ed0fc32797dfa2. Thanks for catching.